src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arr = [];
+  let b;
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      }
+      b = Object.assign({}, newState);
+      arr.push(b);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const key in newState) {
+        if (actions[i].keysToRemove.includes(key)) {
+          delete newState[key];
+        }
+      }
+      b = Object.assign({}, newState);
+      arr.push(b);
```

Comment:
4. [DONT REPEAT YOURSELF] - If you perform same action in all switch cases - do it just once afterwards.
You can `arr.push(b)` at the end of loop. And SWITCH would be better...